algorithms/linfa-nn/src/distance.rs

```diff
@@ -77,6 +77,21 @@ impl<F: Float> Distance<F> for LInfDist {
     }
 }
 
+/// L0 or [Hamming](https://en.wikipedia.org/wiki/Hamming_distance)
+#[derive(Debug, Clone, PartialEq)]
+pub struct L0Dist;
+impl<F: Float> Distance<F> for L0Dist {
+    #[inline]
+    fn distance<D: Dimension>(&self, a: ArrayView<F, D>, b: ArrayView<F, D>) -> F {
+        Zip::from(&a).and(&b).fold(F::zero(), |acc, &a, &b| {
+            if (a - b).is_zero() {
```

Comment:
comparing two floats like this introduces unclear behavior when rounding errors occur

read in order 
    (1) https://stackoverflow.com/questions/5595425/what-is-the-best-way-to-compare-floats-for-almost-equality-in-python 
    (2) https://floating-point-gui.de/errors/comparison/ 
    (3) https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/

and then use the https://docs.rs/approx/latest/approx/ crate to compare by absolute and relative epsilone